cloud: add doc changes for v6.5 (#12041)
pingcap/docs: system-variables.md

### tidb_ddl_enable_fast_reorg <span class="version-mark">New in v6.3.0</span>

<CustomContent platform="tidb-cloud">

> **Note:**
>

> > To improve the speed for index creation using this variable, make sure that your TiDB cluster is hosted on AWS and your TiDB node size is at least 8 vCPU. For [Serverless Tier clusters](/tidb-cloud/select-cluster-tier.md#serverless-tier-beta), this feature is unavailable.

Reviewer: Add this limitation in https://docs.pingcap.com/tidbcloud/serverless-tier-limitations

Author: added in https://github.com/pingcap/docs/pull/12041/commits/e085bbd4198e6faea28b7ff5abc9b7d799ed2d81

### tidb_ddl_enable_fast_reorg <span class="version-mark">New in v6.3.0</span>

<CustomContent platform="tidb-cloud">

> **Note:**
>

> > To improve the speed for index creation using this variable, make sure that your TiDB cluster is hosted on AWS and your TiKV node size is at least 8 vCPU and 16 GiB. For [Serverless Tier clusters](/tidb-cloud/select-cluster-tier.md#serverless-tier-beta), this feature is unavailable.

Author: "your TiKV node size is at least 8 vCPU and 16 GiB" is just a placeholder to be updated

Author: @benmaoer Would you please help confirm? Thanks.

> > To improve the speed for index creation using this variable, make sure that your TiDB cluster is hosted on AWS and your TiKV node size is at least 8 vCPU and 16 GiB. For [Serverless Tier clusters](/tidb-cloud/select-cluster-tier.md#serverless-tier-beta), this feature is unavailable.

Reviewer 2: This isn't available on GCP?

Reviewer 3: Yes, If customers requests it on GCP, it can be supported.

> > To improve the speed for index creation using this variable, make sure that your TiDB cluster is hosted on AWS and your TiKV node size is at least 8 vCPU and 16 GiB. For [Serverless Tier clusters](/tidb-cloud/select-cluster-tier.md#serverless-tier-beta), this feature is unavailable.

Author: ```suggestion
> To improve the speed for index creation using this variable, make sure that your TiDB cluster is hosted on AWS and your TiDB node size is at least 8 vCPU. For [Serverless Tier clusters](/tidb-cloud/select-cluster-tier.md#serverless-tier-beta), this feature is unavailable.
```

### tidb_ddl_enable_fast_reorg <span class="version-mark">New in v6.3.0</span>

<CustomContent platform="tidb-cloud">

> **Note:**
>

> > To improve the speed for index creation using this variable, make sure that your TiDB cluster is hosted on AWS and your TiDB node size is at least 8 vCPU. For [Serverless Tier clusters](/tidb-cloud/select-cluster-tier.md#serverless-tier-beta), this feature is unavailable.

Reviewer: ```suggestion
> To improve the speed for index creation using this variable, make sure that your TiDB cluster is hosted on AWS and your TiDB node size is at least 8 vCPU. For [Serverless Tier](/tidb-cloud/select-cluster-tier.md#serverless-tier-beta) clusters, this feature is unavailable.
```

</CustomContent>

<CustomContent platform="tidb-cloud">

> **Warning:**
>
> Currently, this feature is not fully compatible with [altering multiple columns or indexes in a single `ALTER TABLE` statement](/sql-statements/sql-statement-alter-table.md). When adding a unique index with the index acceleration, you need to avoid altering other columns or indexes in the same statement.
>

> > When [PITR (Point-in-time recovery)](/tidb-cloud/backup-and-restore.md) is disabled, the speed of adding indexes is expected to be about 10 times than that in v6.1.0. However, there is no performance improvement when both PITR and index acceleration are enabled. To optimize performance, it is recommended that you disable PITR, add indexes in a quick way, then enable PITR and perform a full backup. Otherwise, three expected behaviors might occur:

Reviewer 4: ```suggestion
> When [PITR (Point-in-time recovery)](/tidb-cloud/backup-and-restore.md) is disabled, the speed of adding indexes is expected to be about 10 times that in v6.1.0. However, there is no performance improvement when both PITR and index acceleration are enabled. To optimize performance, it is recommended that you disable PITR, add indexes in a quick way, then enable PITR and perform a full backup. Otherwise, these expected behaviors might occur:
```
